query_test.go

```diff
@@ -1,27 +1,77 @@
 package kallax
 
 import (
-	. "gopkg.in/check.v1"
-	"gopkg.in/mgo.v2/bson"
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/suite"
 )
 
-func (s *BaseSuite) TestBaseQuery_AddCriteria(c *C) {
-	q := NewBaseQuery()
-	q.AddCriteria(bson.M{"foo": "foo"})
-	q.AddCriteria(bson.M{"qux": "qux"})
+func TestBaseQuery(t *testing.T) {
+	suite.Run(t, new(QuerySuite))
+}
+
+type QuerySuite struct {
+	suite.Suite
+	q *BaseQuery
+}
+
+func (s *QuerySuite) SetupTest() {
+	s.q = NewBaseQuery("foo")
+}
+
+func (s *QuerySuite) TestSelect() {
+	s.q.Select("a", "b", "c")
+	s.Equal(columnSet{"a", "b", "c"}, s.q.columns)
+}
+
+func (s *QuerySuite) TestSelectNot() {
+	s.q.SelectNot("a", "b", "c")
+	s.Equal(columnSet{"a", "b", "c"}, s.q.excludedColumns)
+}
+
+func (s *QuerySuite) TestCopy() {
+	s.q.Select("a", "b", "c")
+	s.q.SelectNot("foo")
+	s.q.BatchSize(30)
+	s.q.Limit(2)
+	s.q.Offset(30)
+	copy := s.q.Copy()
+
+	s.Equal(s.q, copy)
+	s.NotEqual(unsafe.Pointer(s.q), unsafe.Pointer(copy))
+}
+
+func (s *QuerySuite) TestSelectedColumns() {
+	s.q.Select("a", "b", "c")
+	s.q.SelectNot("b")
+	s.Equal([]string{"a", "c"}, s.q.selectedColumns())
+}
```

Comment:
Add tests for `Select + SelectNot + Select` where at least one of the fields is passed to the three calls. Same for `SelectNot + Select + SelectNot`.